[PT2] set choice handler in config

xuanzhang816 opened the pull request:
Summary:
We were setting the custom inductor choice using `torch._inductor.virtualized.V.set_choices_handler(CustomInductorChoices())`. However, this leads to inconsistent behaviors, even for jobs that are submitted back to back.

In this diff, we pass in the choice handler via an inductor config and overwrite the default behavior when the config is provided. This sovles the inconsistent behavior.

Test Plan: see D85785892 (internal only)

Differential Revision: D85785879




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

xuanzhang816 commented:
@pytorchbot label "topic: not user facing"

eellison commented on torch/_inductor/config.py:
```diff
@@ -653,6 +656,10 @@ def use_autoheuristic(name: str) -> bool:
 )
 
+# Custom InductorChoices class to use, a tuple of (class, kwargs)
+# Example: inductor_choices_class = (MyCustomChoices, {"arg1": value1})
```
nit: get rid of kwargs  here

xuanzhang816 committed:
[PT2] set choice handler in config (#166607)

Summary:

We were setting the custom inductor choice using `torch._inductor.virtualized.V.set_choices_handler(CustomInductorChoices())`. However, this leads to inconsistent behaviors, even for jobs that are submitted back to back.

In this diff, we pass in the choice handler via an inductor config and overwrite the default behavior when the config is provided. This sovles the inconsistent behavior.

Test Plan: see D85785892 (internal only)

Reviewed By: eellison

Differential Revision: D85785879
